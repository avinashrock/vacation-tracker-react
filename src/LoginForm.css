@import url(https://fonts.googleapis.com/css?family=Exo:100,200,400);
@import url(https://fonts.googleapis.com/css?family=Source+Sans+Pro:700,400,300);

body{
	margin: 0;
	padding: 0;
    background: #fff;
    color: #fff;
	font-family: Arial;
	font-size: 12px;
}

.body{
	position: absolute;
	top:85px;
	left: -50px;
	right: -10px;
	bottom:-40px;
	width: auto;
	height: auto;
	/* background-image: url(https://www.tmpresale.com/wp-content/uploads/2015/11/email-newsletters-flying-around-where-presales-really-come-from-300x169.jpg); */
	background-size: cover;
	-webkit-filter: blur(3px);
	z-index: 0;
}

.grad{
	position: absolute;
	top: -20px;
	bottom: -40px;
	width: auto;
	height: auto;
	background: -webkit-gradient(linear, left top, left bottom, color-stop(0%,rgba(0,0,0,0)), color-stop(100%,rgba(0,0,0,0.65))); /* Chrome,Safari4+ */
	z-index: 1;
	opacity: 0.7;
}

.title{
	background-color: black;
	height:150px;
	width:1590px;
	font-size:30px;
	color:white;
	padding-left:640px;
	padding-top:20px;
	list-style-type: none;
	margin-left: -50px;
}



.login{
	position: absolute;
	top: calc(50% - 95px);
	left: calc(50% - 130px);
	height: 320px;
	width: 300px;
	padding: 5px;
	z-index: 2;
    background-color:black;
}

.login input[type=text]{
	width: 290px;
	height: 50px;
	background: transparent;
	border: 1px solid grey;
	border-radius: 2px;
	color: black;
	font-family: 'Exo', sans-serif;
	font-size: 16px;
	font-weight: 400;
	padding-top: 15px;
  	padding-left: 10px;
  	margin-top: 30px;
    background-color: white;
}

.login input[type=password]{
	width: 290px;
	height: 50px;
	background: transparent;
	border: 1px solid grey;
	border-radius: 2px;
	color:black;
	font-family: 'Exo', sans-serif;
	font-size: 16px;
	font-weight: 400;
	padding-left: 10px;
    background-color: white;
	margin-top: 20px;

}

.login input[type=button]{
	width: 290px;
	height: 50px;
	background: #0D94D2;
	border: 1px solid black;
	cursor: pointer;
	border-radius: 2px;
	color: white;
	font-family: 'Exo', sans-serif;
	font-size: 16px;
	font-weight: 400;
	padding: 6px;
	margin-top: 30px;
}

.login input[type=button]:hover{
	opacity: 0.8;
}

.login input[type=button]:active{
	opacity: 0.6;
}

.login input[type=text]:focus{
	outline: none;
	border: 1px solid rgba(255,255,255,0.9);
    color:black;
}

.login input[type=password]:focus{
	outline: none;
	border: 1px solid rgba(255,255,255,0.9);
}

.login input[type=button]:focus{
	outline: none;
}

::-webkit-input-placeholder{
   color: rgba(255,255,255,0.6);
   color:black;
}

::-moz-input-placeholder{
   color: rgba(255,255,255,0.6);
   color:black;
}
.loginButton{
	padding-top:25px;
	background-color: black;
}
h1{
    padding-left:500px;
    font-weight: bold;
    padding-top: 80px;
    color:black;
  }