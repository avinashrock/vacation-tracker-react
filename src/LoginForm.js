import React, { Component } from 'react';
import './LoginForm.css';
import axios from 'axios';
import Modal from 'react-responsive-modal';
import { darkBlack } from 'material-ui/styles/colors';

let center = {
    textAlign: 'center',
    color:darkBlack,
    fontSize:75,
}
class LoginForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            loggedIn: '',
            open: false,
        }
        this.handleClick = this.handleClick.bind(this);
        this.onCloseModal = this.onCloseModal.bind(this);
        

    }
    componentWillMount(){
        const {history} = this.props; 
        if(sessionStorage.getItem('username') !== null && sessionStorage.getItem('username') !== ''){
        if(sessionStorage.getItem('reportee') === "true")
            history.push('/ManagerLeave');
        else {
            history.push('/AssociateLeave');
        }
    }
    }
    onChange = (e) =>{
        this.setState({
            [e.target.name]: e.target.value
        })
    }
    onCloseModal() {
        this.setState({ open: false });
    }
    handleClick(event) {
        event.preventDefault();
        const { history } = this.props;
        this.setState({ loggedIn: false });
        axios.post('http://localhost:8080/service/Vacation-Tracker/login', { username: this.refs.username.value, password: this.refs.password.value })
            .then(response => {
                sessionStorage.setItem("username",this.refs.username.value);
                sessionStorage.setItem("name",response.data.name);
                sessionStorage.setItem("reportee",response.data.reportee);

                if ((response.status === 200 && response.data.reportee === false)) {
                    this.setState({ loggedIn: true });
                    history.push({
                        pathname: '/AssociateLeave',
                        state: { detail: response.data }
                    })
                }
                else if ((response.status === 200 && response.data.reportee === true)) {
                    this.setState({ loggedIn: true });
                    history.push({
                        pathname: '/ManagerLeave',
                        state: { detail: response.data }
                    })
                }
            }, error => {
                this.setState({ open: true });
            }
            ).catch(error => {
                alert("internal server error");
            }
        );
    }

    render(){
       return (
        <React.Fragment>
            <div>
            <h3 style={center}>
                Vacation Tracker</h3>
           </div>
                <div>
                    <div className="body"></div>
                    <div className="grad"></div>
                    <br />
                    <div className="login" >
                    
                        <form action="" onSubmit={(event) => this.handleClick(event)}>
                            <input type="text" ref="username" placeholder="AssociateId" name="username" onChange={this.onChange} required /><br />
                            <input type="password" ref="password" placeholder="Password" name="password " onChange={this.onChange} required /><br />
                            <div className="loginButton">
                                <button type="submit" className="submit" >Assocaite Login</button>
                            </div>
                        </form> 
                    </div>
                </div>
                <div>
                    <Modal open={this.state.open} onClose={this.onCloseModal} center>
                        <h3>Unauthorized Client error</h3>
                    </Modal>
                </div>
            </React.Fragment>
        );
    }

}
export default LoginForm;