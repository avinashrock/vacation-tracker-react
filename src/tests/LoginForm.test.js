import React from 'react';
import {shallow} from 'enzyme';
import LoginForm from '../LoginForm';


const historyMock = { push: jest.fn() };
const mockEvent = { preventDefault: jest.fn() };
const wrapper = shallow(<LoginForm history={historyMock} />);
describe('Component should render without crashing',()=>{
    it('Component renders correctly',()=>{
        expect(wrapper.exists()).toBe(true);
    })
    it('Component matches its snapshot',()=>{
        expect(wrapper).toMatchSnapshot();
    })
    it('should handle state changes ', () => {
      
        expect(wrapper.state().open).toEqual(false);
        wrapper.simulate('click');
        expect(wrapper.state().open).toEqual(false);
    })
    it('should handle state changes', () => {
        
        expect(wrapper.state().loggedIn).toEqual('');
        wrapper.simulate('click',{target:{loggedIn:false}});
        
    })
    it('should handle state changes', () => {
      
        expect(wrapper.state().username).toBe('');
        wrapper.simulate('change',{target: {username:'avinash'}});
    })
    
    it('should handle state chnges', () =>{
      
        expect(wrapper.state().password).toBe('');
        wrapper.simulate('change',{target:{password:'rocks'}});
    })

    // it('handleClick',()=>{
    //     // wrapper.instance().handleClick();
    //     const form = wrapper.find('form'); 
    //     form.simulate('submit', mockEvent);
    // })
    it('onCloseModal',()=>{
        wrapper.instance().onCloseModal();
    })
    // it('onChange',()=>{
    //     wrapper.instance().onChange();
    // })
    it('className body exists',()=>{
        expect(wrapper.find('.body')).toHaveLength(1);
    })
    it('className grad exists',()=>{
        expect(wrapper.find('.grad')).toHaveLength(1);
    })
    it('className login exists',()=>{
        expect(wrapper.find('.login')).toHaveLength(1);
    })
    it('className loginButton exists',()=>{
        expect(wrapper.find('.loginButton')).toHaveLength(1);
    })
    it('input change for username',()=>{
        const input = wrapper.find('input').at(0);
        input.simulate('change', { target: { value: 'Changed' } });
    })
    it('input change for password',()=>{
        const input = wrapper.find('input').at(1);
        input.simulate('change', { target: { value: 'Changed' } });
    })
    it('button with className submit',()=>{
        const button = wrapper.find('button');
        expect(button.exists()).toBe(true);
    })
    it('lifecycle method renders correctly',()=>{
        wrapper.instance().componentWillMount();
    })
    // it('simulate form  submit',()=>{
    //     wrapper.find('form').simulate('submit', { preventDefault () {} },);
    // })
})