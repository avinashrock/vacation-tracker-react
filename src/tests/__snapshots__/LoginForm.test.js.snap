// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should render without crashing Component matches its snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div>
          <h3
                    style={
                              Object {
                                        "color": "rgba(0, 0, 0, 0.87)",
                                        "fontSize": 75,
                                        "textAlign": "center",
                                      }
                    }
          >
                    Vacation Tracker
          </h3>
</div>,
        <div>
          <div
                    className="body"
          />
          <div
                    className="grad"
          />
          <br />
          <div
                    className="login"
          >
                    <form
                              action=""
                              onSubmit={[Function]}
                    >
                              <input
                                        name="username"
                                        onChange={[Function]}
                                        placeholder="AssociateId"
                                        required={true}
                                        type="text"
                              />
                              <br />
                              <input
                                        name="password "
                                        onChange={[Function]}
                                        placeholder="Password"
                                        required={true}
                                        type="password"
                              />
                              <br />
                              <div
                                        className="loginButton"
                              >
                                        <button
                                                  className="submit"
                                                  type="submit"
                                        >
                                                  Assocaite Login
                                        </button>
                              </div>
                    </form>
          </div>
</div>,
        <div>
          <Modal
                    animationDuration={500}
                    center={true}
                    classNames={Object {}}
                    classes={
                              Object {
                                        "closeButton": "styles_closeButton__20ID4",
                                        "closeIcon": "styles_closeIcon__1QwbI",
                                        "modal": "styles_modal__gNwvD",
                                        "overlay": "styles_overlay__CLSq-",
                                        "overlayCenter": "styles_overlayCenter__YHoO7",
                                        "transitionEnter": "styles_transitionEnter__3j_-a",
                                        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                                        "transitionExit": "styles_transitionExit__1KmEf",
                                        "transitionExitActive": "styles_transitionExitActive__1nQXw",
                                      }
                    }
                    closeIconSize={28}
                    closeIconSvgPath={
                              <path
                                        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                              />
                    }
                    closeOnEsc={true}
                    closeOnOverlayClick={true}
                    onClose={[Function]}
                    onEntered={null}
                    onEscKeyDown={null}
                    onExited={null}
                    onOverlayClick={null}
                    open={false}
                    showCloseIcon={true}
                    styles={Object {}}
          >
                    <h3>
                              Unauthorized Client error
                    </h3>
          </Modal>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3
            style={
                        Object {
                                    "color": "rgba(0, 0, 0, 0.87)",
                                    "fontSize": 75,
                                    "textAlign": "center",
                                  }
            }
>
            Vacation Tracker
</h3>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vacation Tracker",
            "style": Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": 75,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Vacation Tracker",
          "type": "h3",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="body"
/>,
            <div
              className="grad"
/>,
            <br />,
            <div
              className="login"
>
              <form
                            action=""
                            onSubmit={[Function]}
              >
                            <input
                                          name="username"
                                          onChange={[Function]}
                                          placeholder="AssociateId"
                                          required={true}
                                          type="text"
                            />
                            <br />
                            <input
                                          name="password "
                                          onChange={[Function]}
                                          placeholder="Password"
                                          required={true}
                                          type="password"
                            />
                            <br />
                            <div
                                          className="loginButton"
                            >
                                          <button
                                                        className="submit"
                                                        type="submit"
                                          >
                                                        Assocaite Login
                                          </button>
                            </div>
              </form>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "body",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "grad",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                action=""
                onSubmit={[Function]}
>
                <input
                                name="username"
                                onChange={[Function]}
                                placeholder="AssociateId"
                                required={true}
                                type="text"
                />
                <br />
                <input
                                name="password "
                                onChange={[Function]}
                                placeholder="Password"
                                required={true}
                                type="password"
                />
                <br />
                <div
                                className="loginButton"
                >
                                <button
                                                className="submit"
                                                type="submit"
                                >
                                                Assocaite Login
                                </button>
                </div>
</form>,
              "className": "login",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "action": "",
                "children": Array [
                  <input
                    name="username"
                    onChange={[Function]}
                    placeholder="AssociateId"
                    required={true}
                    type="text"
/>,
                  <br />,
                  <input
                    name="password "
                    onChange={[Function]}
                    placeholder="Password"
                    required={true}
                    type="password"
/>,
                  <br />,
                  <div
                    className="loginButton"
>
                    <button
                                        className="submit"
                                        type="submit"
                    >
                                        Assocaite Login
                    </button>
</div>,
                ],
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "username",
                    "onChange": [Function],
                    "placeholder": "AssociateId",
                    "required": true,
                    "type": "text",
                  },
                  "ref": "username",
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "password ",
                    "onChange": [Function],
                    "placeholder": "Password",
                    "required": true,
                    "type": "password",
                  },
                  "ref": "password",
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="submit"
                      type="submit"
>
                      Assocaite Login
</button>,
                    "className": "loginButton",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Assocaite Login",
                      "className": "submit",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Assocaite Login",
                    "type": "button",
                  },
                  "type": "div",
                },
              ],
              "type": "form",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Modal
            animationDuration={500}
            center={true}
            classNames={Object {}}
            classes={
                        Object {
                                    "closeButton": "styles_closeButton__20ID4",
                                    "closeIcon": "styles_closeIcon__1QwbI",
                                    "modal": "styles_modal__gNwvD",
                                    "overlay": "styles_overlay__CLSq-",
                                    "overlayCenter": "styles_overlayCenter__YHoO7",
                                    "transitionEnter": "styles_transitionEnter__3j_-a",
                                    "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                                    "transitionExit": "styles_transitionExit__1KmEf",
                                    "transitionExitActive": "styles_transitionExitActive__1nQXw",
                                  }
            }
            closeIconSize={28}
            closeIconSvgPath={
                        <path
                                    d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                        />
            }
            closeOnEsc={true}
            closeOnOverlayClick={true}
            onClose={[Function]}
            onEntered={null}
            onEscKeyDown={null}
            onExited={null}
            onOverlayClick={null}
            open={false}
            showCloseIcon={true}
            styles={Object {}}
>
            <h3>
                        Unauthorized Client error
            </h3>
</Modal>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animationDuration": 500,
            "center": true,
            "children": <h3>
              Unauthorized Client error
</h3>,
            "classNames": Object {},
            "classes": Object {
              "closeButton": "styles_closeButton__20ID4",
              "closeIcon": "styles_closeIcon__1QwbI",
              "modal": "styles_modal__gNwvD",
              "overlay": "styles_overlay__CLSq-",
              "overlayCenter": "styles_overlayCenter__YHoO7",
              "transitionEnter": "styles_transitionEnter__3j_-a",
              "transitionEnterActive": "styles_transitionEnterActive___eQs7",
              "transitionExit": "styles_transitionExit__1KmEf",
              "transitionExitActive": "styles_transitionExitActive__1nQXw",
            },
            "closeIconSize": 28,
            "closeIconSvgPath": <path
              d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
/>,
            "closeOnEsc": true,
            "closeOnOverlayClick": true,
            "onClose": [Function],
            "onEntered": null,
            "onEscKeyDown": null,
            "onExited": null,
            "onOverlayClick": null,
            "open": false,
            "showCloseIcon": true,
            "styles": Object {},
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Unauthorized Client error",
            },
            "ref": null,
            "rendered": "Unauthorized Client error",
            "type": "h3",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            <h3
                        style={
                                    Object {
                                                "color": "rgba(0, 0, 0, 0.87)",
                                                "fontSize": 75,
                                                "textAlign": "center",
                                              }
                        }
            >
                        Vacation Tracker
            </h3>
</div>,
          <div>
            <div
                        className="body"
            />
            <div
                        className="grad"
            />
            <br />
            <div
                        className="login"
            >
                        <form
                                    action=""
                                    onSubmit={[Function]}
                        >
                                    <input
                                                name="username"
                                                onChange={[Function]}
                                                placeholder="AssociateId"
                                                required={true}
                                                type="text"
                                    />
                                    <br />
                                    <input
                                                name="password "
                                                onChange={[Function]}
                                                placeholder="Password"
                                                required={true}
                                                type="password"
                                    />
                                    <br />
                                    <div
                                                className="loginButton"
                                    >
                                                <button
                                                            className="submit"
                                                            type="submit"
                                                >
                                                            Assocaite Login
                                                </button>
                                    </div>
                        </form>
            </div>
</div>,
          <div>
            <Modal
                        animationDuration={500}
                        center={true}
                        classNames={Object {}}
                        classes={
                                    Object {
                                                "closeButton": "styles_closeButton__20ID4",
                                                "closeIcon": "styles_closeIcon__1QwbI",
                                                "modal": "styles_modal__gNwvD",
                                                "overlay": "styles_overlay__CLSq-",
                                                "overlayCenter": "styles_overlayCenter__YHoO7",
                                                "transitionEnter": "styles_transitionEnter__3j_-a",
                                                "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                                                "transitionExit": "styles_transitionExit__1KmEf",
                                                "transitionExitActive": "styles_transitionExitActive__1nQXw",
                                              }
                        }
                        closeIconSize={28}
                        closeIconSvgPath={
                                    <path
                                                d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                                    />
                        }
                        closeOnEsc={true}
                        closeOnOverlayClick={true}
                        onClose={[Function]}
                        onEntered={null}
                        onEscKeyDown={null}
                        onExited={null}
                        onOverlayClick={null}
                        open={false}
                        showCloseIcon={true}
                        styles={Object {}}
            >
                        <h3>
                                    Unauthorized Client error
                        </h3>
            </Modal>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3
              style={
                            Object {
                                          "color": "rgba(0, 0, 0, 0.87)",
                                          "fontSize": 75,
                                          "textAlign": "center",
                                        }
              }
>
              Vacation Tracker
</h3>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vacation Tracker",
              "style": Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontSize": 75,
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": "Vacation Tracker",
            "type": "h3",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="body"
/>,
              <div
                className="grad"
/>,
              <br />,
              <div
                className="login"
>
                <form
                                action=""
                                onSubmit={[Function]}
                >
                                <input
                                                name="username"
                                                onChange={[Function]}
                                                placeholder="AssociateId"
                                                required={true}
                                                type="text"
                                />
                                <br />
                                <input
                                                name="password "
                                                onChange={[Function]}
                                                placeholder="Password"
                                                required={true}
                                                type="password"
                                />
                                <br />
                                <div
                                                className="loginButton"
                                >
                                                <button
                                                                className="submit"
                                                                type="submit"
                                                >
                                                                Assocaite Login
                                                </button>
                                </div>
                </form>
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "body",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "grad",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <form
                  action=""
                  onSubmit={[Function]}
>
                  <input
                                    name="username"
                                    onChange={[Function]}
                                    placeholder="AssociateId"
                                    required={true}
                                    type="text"
                  />
                  <br />
                  <input
                                    name="password "
                                    onChange={[Function]}
                                    placeholder="Password"
                                    required={true}
                                    type="password"
                  />
                  <br />
                  <div
                                    className="loginButton"
                  >
                                    <button
                                                      className="submit"
                                                      type="submit"
                                    >
                                                      Assocaite Login
                                    </button>
                  </div>
</form>,
                "className": "login",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "action": "",
                  "children": Array [
                    <input
                      name="username"
                      onChange={[Function]}
                      placeholder="AssociateId"
                      required={true}
                      type="text"
/>,
                    <br />,
                    <input
                      name="password "
                      onChange={[Function]}
                      placeholder="Password"
                      required={true}
                      type="password"
/>,
                    <br />,
                    <div
                      className="loginButton"
>
                      <button
                                            className="submit"
                                            type="submit"
                      >
                                            Assocaite Login
                      </button>
</div>,
                  ],
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "name": "username",
                      "onChange": [Function],
                      "placeholder": "AssociateId",
                      "required": true,
                      "type": "text",
                    },
                    "ref": "username",
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "name": "password ",
                      "onChange": [Function],
                      "placeholder": "Password",
                      "required": true,
                      "type": "password",
                    },
                    "ref": "password",
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <button
                        className="submit"
                        type="submit"
>
                        Assocaite Login
</button>,
                      "className": "loginButton",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Assocaite Login",
                        "className": "submit",
                        "type": "submit",
                      },
                      "ref": null,
                      "rendered": "Assocaite Login",
                      "type": "button",
                    },
                    "type": "div",
                  },
                ],
                "type": "form",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Modal
              animationDuration={500}
              center={true}
              classNames={Object {}}
              classes={
                            Object {
                                          "closeButton": "styles_closeButton__20ID4",
                                          "closeIcon": "styles_closeIcon__1QwbI",
                                          "modal": "styles_modal__gNwvD",
                                          "overlay": "styles_overlay__CLSq-",
                                          "overlayCenter": "styles_overlayCenter__YHoO7",
                                          "transitionEnter": "styles_transitionEnter__3j_-a",
                                          "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                                          "transitionExit": "styles_transitionExit__1KmEf",
                                          "transitionExitActive": "styles_transitionExitActive__1nQXw",
                                        }
              }
              closeIconSize={28}
              closeIconSvgPath={
                            <path
                                          d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                            />
              }
              closeOnEsc={true}
              closeOnOverlayClick={true}
              onClose={[Function]}
              onEntered={null}
              onEscKeyDown={null}
              onExited={null}
              onOverlayClick={null}
              open={false}
              showCloseIcon={true}
              styles={Object {}}
>
              <h3>
                            Unauthorized Client error
              </h3>
</Modal>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "animationDuration": 500,
              "center": true,
              "children": <h3>
                Unauthorized Client error
</h3>,
              "classNames": Object {},
              "classes": Object {
                "closeButton": "styles_closeButton__20ID4",
                "closeIcon": "styles_closeIcon__1QwbI",
                "modal": "styles_modal__gNwvD",
                "overlay": "styles_overlay__CLSq-",
                "overlayCenter": "styles_overlayCenter__YHoO7",
                "transitionEnter": "styles_transitionEnter__3j_-a",
                "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                "transitionExit": "styles_transitionExit__1KmEf",
                "transitionExitActive": "styles_transitionExitActive__1nQXw",
              },
              "closeIconSize": 28,
              "closeIconSvgPath": <path
                d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
/>,
              "closeOnEsc": true,
              "closeOnOverlayClick": true,
              "onClose": [Function],
              "onEntered": null,
              "onEscKeyDown": null,
              "onExited": null,
              "onOverlayClick": null,
              "open": false,
              "showCloseIcon": true,
              "styles": Object {},
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Unauthorized Client error",
              },
              "ref": null,
              "rendered": "Unauthorized Client error",
              "type": "h3",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
